eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -335,9 +336,14 @@ def insert_atomic_to_struct(self, atomic, struct):
         Adds or replaces atomic subtree in struct tree.
         """
         if len(struct.children) == 2:
+            for i in range(len(struct.children[1].children)):
+                if self.get_name(atomic) == self.get_name(
+                    struct.children[1].children[i]
+                ):
+                    raise ComplexParsingError("Matching agent was not found", struct)
```

Comment:
I would go with something like:

```suggestion
                    raise ComplexParsingError(f"Illegal nesting sequence: {atomic}:{struct}")
```

Reply:
Thank you, exceptions are edited now.